lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -22,25 +22,16 @@ use constant DEPENDENCY_DEBUG_INFO => $ENV{OPENQA_DEPENDENCY_DEBUG_INFO};
 
 sub referer_check ($self) {
     return $self->reply->not_found if (!defined $self->param('testid'));
-    my $referer = $self->req->headers->header('Referer') // '';
-    return 1 unless $referer;
+    return undef unless my $referer = $self->req->headers->header('Referer') // '';
     $self->schema->resultset('Jobs')->mark_job_linked($self->param('testid'), $referer);
     return 1;
 }
 
-sub list {
-    my ($self) = @_;
-}
-
-sub get_match_param {
-    my ($self) = @_;
+sub list ($self) { }
 
-    my $match;
-    if (defined($self->param('match'))) {
-        $match = $self->param('match');
-        $match =~ s/[^\w\[\]\{\}\(\),:.+*?\\\$^|-]//g;    # sanitize
-    }
-    return $match;
+sub get_match_param ($self) {
+    return unless my $match = $self->param('match');
```

Comment:
```suggestion
    return undef unless my $match = $self->param('match');
```

We should always return something. If I'm reading it correctly `undef` is what we want here.

Reply:
yes, and looking at the usage this is very important here because we use the method in list context